forms-flow-review/src/components/TaskFilterModal/TaskFilterModalBody.tsx

```diff
@@ -212,8 +216,8 @@ const TaskFilterModalBody = ({
     const { roles, users, criteria, properties } = filterToEdit;
     const { assignee, sorting, candidateGroup } = criteria;
     setShareFilterForSpecificRole(roles);
-    setAccessOption(assignee ? "specificAssignee" : "specificRole");
-    setAccessValue(assignee ? assignee : candidateGroup);
+    setAccessOption(assignee ? "specificAssignee" : candidateGroup? "specificRole":"currentUser");
```

Comment:
the condition check is looks good, but small improvement needed here like we can declare a constant variable top of the code and use it  instead of the string 
const SPECIFIC_ROLE="specificRole" like this
"specificAssignee" , "specificRole", "currentUser",
and use the variable across the code